code/controllers/subsystem/fluids/fluid_effects.dm

```diff
@@ -0,0 +1,83 @@
+/**
+ *
+ */
+SUBSYSTEM_DEF(fluid_effect)
+	name = "Fluid Effect"
+	wait = 1
+	flags = SS_BACKGROUND|SS_POST_FIRE_TIMING
+
+	//
+	/// The amount of time
+	var/effect_wait = 1 SECONDS
+	///
+	var/list/list/obj/effect/particle_effect/fluid/processing
+	///
+	var/tmp/list/obj/effect/particle_effect/fluid/currentrun
+	///
+	var/tmp/num_buckets
+	///
+	var/tmp/processing_bucket_index
```

Comment:
I uh, I think you're missing some comments here

Reply:
Yeah, it's it a bit of a temporary state until either this is settled as the final version or effects and spreading get merged.

Reply:
I've filled out the comments.